packages/llamaindex/src/storage/vectorStore/UpstashVectorStore.ts

```diff
@@ -0,0 +1,203 @@
+   * @param embeddingResults The Nodes to be inserted, optionally including metadata tuples.
+   * @returns Due to limitations in the Upstash client, does not return the upserted ID list, only a Promise resolve/reject.
+   */
+  async add(embeddingResults: BaseNode<Metadata>[]): Promise<string[]> {
+    if (embeddingResults.length == 0) {
+      return Promise.resolve([]);
+    }
+
+    const db = this.db;
+    const nodes = embeddingResults.map(this.nodeToRecord);
+    const result = await db.upsert(nodes, { namespace: this.namespace });
+    if (result != "OK") {
+      return Promise.reject(new Error("Failed to save chunk"));
+    }
+    return Promise.resolve([]);
+  }
+
+  /**
+   * Adds plain text record(s) to the table. Upstash take cares of embedding conversion.
+   * @param embeddingResults The Nodes to be inserted, optionally including metadata tuples.
+   * @returns Due to limitations in the Upstash client, does not return the upserted ID list, only a Promise resolve/reject.
+   */
+  async addPlainText(
+    embeddingResults: TextNode<Metadata>[],
+  ): Promise<string[]> {
+    if (embeddingResults.length == 0) {
+      return Promise.resolve([]);
+    }
+
+    const db = this.db;
+    const nodes = embeddingResults.map(this.textNodeToRecord);
+    const result = await db.upsert(nodes, { namespace: this.namespace });
+    if (result != "OK") {
+      return Promise.reject(new Error("Failed to save chunk"));
+    }
+    return Promise.resolve([]);
+  }
+
+  /**
+   * Deletes a single record from the database by id.
+   * NOTE: Uses the collection property controlled by setCollection/getCollection.
+   * @param refDocId Unique identifier for the record to delete.
+   * @returns Promise that resolves if the delete query did not throw an error.
+   */
+  async delete(refDocId: string): Promise<void> {
+    await this.db.namespace(this.namespace).delete(refDocId);
+    return Promise.resolve();
+  }
+
+  /**
+   * Deletes a single record from the database by id.
+   * NOTE: Uses the collection property controlled by setCollection/getCollection.
+   * @param refDocId Unique identifier for the record to delete.
+   * @param deleteKwargs Required by VectorStore interface.  Currently ignored.
+   * @returns Promise that resolves if the delete query did not throw an error.
+   */
+  async deleteMany(refDocId: string[]): Promise<void> {
+    await this.db.namespace(this.namespace).delete(refDocId);
+    return Promise.resolve();
+  }
+
+  /**
+   * Query the vector store for the closest matching data to the query embeddings
+   * @param query The VectorStoreQuery to be used
+   * @param options Required by VectorStore interface.  Currently ignored.
+   * @returns Zero or more Document instances with data from the vector store.
+   */
+  async query(
+    query: VectorStoreQuery,
+    _options?: any,
+  ): Promise<VectorStoreQueryResult> {
+    const filter = this.toUpstashFilter(query.filters);
+
+    const defaultOptions: any = {
```

Comment:
I guess we can get the type from sdk, instead of any

Reply:
I think its safe to keep it like this since its internal.